ledger-core/src/CMakeLists.txt

```diff
@@ -102,6 +100,11 @@ else()
     include(StaticAnalyzers)
     set_enabled_static_analyzers(ledger-core-obj)
 
+    # set compiler warnings
+    include(CompilerWarnings)
+    set_compiler_warnings(ledger-core-obj)
+    disable_compiler_warnings(ledger-core-api-obj)
```

Comment:
Yeah see, this why `set` / `enable` are not good names. It feels like you always disable warnings for API (which I think is okay) and always set warnings for our code. I’m not feeling comfortable with that naming, please find other names (`toggle` is still a good candidate to me).